tools/checklocks/analysis.go

```diff
@@ -497,32 +497,15 @@ func (pc *passContext) checkInstruction(inst ssa.Instruction, ls *lockState) (*s
 	return nil, nil
 }
 
-// checkBasicBlock traverses the control flow graph starting at a set of given
-// block and checks each instruction for allowed operations.
-func (pc *passContext) checkBasicBlock(fn *ssa.Function, block *ssa.BasicBlock, lff *lockFunctionFacts, parent *lockState, seen map[*ssa.BasicBlock]*lockState) *lockState {
-	if oldLS, ok := seen[block]; ok && oldLS.isCompatible(parent) {
-		return nil
-	}
-
-	// If the lock state is not compatible, then we need to do the
-	// recursive analysis to ensure that it is still sane. For example, the
-	// following is guaranteed to generate incompatible locking states:
-	//
-	//	if foo {
-	//		mu.Lock()
-	//	}
-	//	other stuff ...
-	//	if foo {
-	//		mu.Unlock()
-	//	}
-
+// checkBasicBlock checks each instruction of basic block for allowed operations.
+// The exit lock state of the basic block is returned if the block does not return, otherwise nil is returned.
```

Comment:
This should be `The exit lock state of the basic block is returned if the block does not return, otherwise nil is returned`